drivers/spi/spi_nrfx_spi.c

```diff
@@ -321,8 +321,26 @@ static int spi_nrfx_pm_control(struct device *dev, u32_t ctrl_command,
 }
 #endif /* CONFIG_DEVICE_POWER_MANAGEMENT */
 
+#define SPI_NRFX_MISO_PULL_DOWN(idx) \
+	IS_ENABLED(DT_NORDIC_NRF_SPI_SPI_##idx##_MISO_PULL_DOWN)
+
+#define SPI_NRFX_MISO_PULL_UP(idx) \
+	IS_ENABLED(DT_NORDIC_NRF_SPI_SPI_##idx##_MISO_PULL_UP)
+
+#define SPI_NRFX_MISO_PULL(idx)				   \
+	COND_CODE_1(SPI_NRFX_MISO_PULL_UP(idx),		   \
```

Comment:
Maybe a ternary operator here is enough?

Reply:
I don't think I understand your suggestion. Could you elaborate a bit on that, please?

Reply:
I think we don't need such elaborate macro where standard operators do just fine.
Something like this should work:
```
SPI_NRFX_MISO_PULL_UP(idx) 
    ? SPI_NRFX_MISO_PULL_DOWN(idx) ? -1 : NRF_GPIO_PIN_PULLUP
    : SPI_NRFX_MISO_PULL_DOWN(idx) ? NRF_GPIO_PIN_PULLDOWN : NRF_GPIO_PIN_NOPULL
```
But its a matter of preference

Reply:
Now I see. And I think you're right. I overcomplicated this. No need to use `COND_CODE_1` here.